lib/install.go

```diff
@@ -51,17 +50,31 @@ func GetInstallLocation(installPath string) string {
 }
 
 // install : install the provided version in the argument
-func install(product Product, tfversion string, binPath string, installPath string, mirrorURL string) {
+func install(product Product, tfversion string, binPath string, installPath string, mirrorURL string) error {
 	var wg sync.WaitGroup
-	/* Check to see if user has permission to the default bin location which is  "/usr/local/bin/terraform"
-	 * If user does not have permission to default bin location, proceed to create $HOME/bin and install the tfswitch there
-	 * Inform user that they don't have permission to default location, therefore tfswitch was installed in $HOME/bin
-	 * Tell users to add $HOME/bin to their path
-	 */
-	binPath = installableBinLocation(product, binPath)
 
-	initialize(binPath)                               //initialize path
-	installLocation = GetInstallLocation(installPath) //get installation location -  this is where we will put our terraform binary file
+	//check to see if the requested version has been downloaded before
+	installLocation := GetInstallLocation(installPath)
+	installFileVersionPath := ConvertExecutableExt(filepath.Join(installLocation, product.GetVersionPrefix()+tfversion))
+	recentDownloadFile := CheckFileExist(installFileVersionPath)
+	if recentDownloadFile {
+		err := ChangeProductSymlink(product, installFileVersionPath, binPath)
```

Comment:
This logic appears identical to L113, I wonder if it worth removing the duplication, e.g.
```
func install() {
    installLocation := GetInstallLocation(installPath)
    installFileVersionPath := ConvertExecutableExt(filepath.Join(installLocation, product.GetVersionPrefix()+tfversion))
    recentDownloadFile := CheckFileExist(installFileVersionPath)
    if !recentDownloadFile {
        tfList := getTfList()
        ... remaining download logic
    }
    err := ChangeProductSymlink()
    ...
}
```

Reply:
I've added `func switchToVersion(product Product, tfversion string, binPath string, installPath string, installFileVersionPath string) error`